packages/cardpay-sdk/sdk/prepaid-card/base.ts

```diff
@@ -388,22 +434,39 @@ export default class PrepaidCard {
     };
   }
 
+  async create(txnHash: string): Promise<{ prepaidCards: PrepaidCardSafe[]; txnReceipt: TransactionReceipt }>;
   async create(
     safeAddress: string,
     tokenAddress: string,
     faceValues: number[],
     customizationDID: string | undefined,
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
+  ): Promise<{ prepaidCards: PrepaidCardSafe[]; txnReceipt: TransactionReceipt }>;
+  async create(
+    safeAddressOrTxnHash: string,
+    tokenAddress?: string,
+    faceValues?: number[],
+    customizationDID?: string | undefined,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<{ prepaidCards: PrepaidCardSafe[]; txnReceipt: TransactionReceipt }> {
-    if (faceValues.length > MAX_PREPAID_CARD_AMOUNT) {
+    if (!tokenAddress) {
+      let txnHash = safeAddressOrTxnHash;
+      return {
+        prepaidCards: await this.resolvePrepaidCards(await this.getPrepaidCardsFromTxn(txnHash)),
+        txnReceipt: await waitUntilTransactionMined(this.layer2Web3, txnHash),
+      };
+    }
+    let safeAddress = safeAddressOrTxnHash;
+    if (faceValues!.length > MAX_PREPAID_CARD_AMOUNT) {
```

Comment:
i'd suggest just checking for faceValues and throwing if they don't exist instead of using `!` like my comment above.